### PR TITLE
[AppConfig] Not sharing the same default http pipeline instance between AppConfig clients + Prepare Sep Release

### PR DESCRIPTION
**Pull request by mssfang**
- fixes: https://github.com/Azure/azure-sdk-for-java/issues/30536
  Each new instance of the AppConfig client without a custom pipeline provided should not share the same default HTTP pipeline.

- Prepare for the Patch release in September.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 module com.azure.data.appconfiguration {
     requires transitive com.azure.core;
+    requires transitive com.azure.http.netty;
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java:6]: Netty should not be added as the user can opt to use a different HTTP client.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
+    requires java.sql;
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java:7]: Why do we need java.sql?

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java:7]: `java.sql` is required to compile samples. I also agree that this isn't the right location for this and should instead investigate making the test `module-info` that was removed in the PR work better with IDEs like IntelliJ

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java (hunk 1 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java]: will revert the changes

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java (hunk 2 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/module-info.java]: will revert the changes

**Review by srnagar — APPROVED** [review]

**Review by samvaity — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**